perceptrain/callbacks/callback.py

```diff
@@ -666,6 +666,134 @@ def run_callback(self, trainer: Any, config: TrainConfig, writer: BaseWriter) ->
+                    gamma=0.5,
+                    threshold=1e-4,
+                    min_lr=1e-5,
+                )
+
+    config = TrainConfig(
+        max_iter=10000,
+        # Print metrics every 1000 training epochs
+        print_every=1000,
+        # Add the custom callback
+        callbacks=[lr_plateau]
+    )
+    ```
+    """
+
+    def __init__(
+        self,
+        on: str,
+        called_every: int = 1,
+        monitor: str = "train_loss",
+        patience: int = 20,
+        mode: str = "min",
+        gamma: float = 0.5,
+        threshold: float = 1e-4,
+        min_lr: float = 1e-6,
+        verbose: bool = True,
+    ):
+        """Initializes the LRSchedulerReduceOnPlateau callback.
+
+        Args:
+            on (str): The event to trigger the callback. Default is `train_epoch_end`.
+            called_every (int, optional): Frequency of callback calls in terms of iterations.
+                Default is 1.
+            monitor (str, optional): The metric to monitor (e.g., "val_loss" or "train_loss").
+                All metrics returned by optimize step are available to monitor.
+                Please add "val_" and "train_" strings at the start of the metric name.
+                Default is "train_loss".
+            mode (str, optional): Whether to minimize ("min") or maximize ("max") the metric.
+                Default is "min".
+            patience (int, optional): Number of allowed iterations with no loss
+                improvement before reducing the learning rate. Default is 20.
+            gamma (float, optional): The decay factor applied to the learning rate. A value
+                < 1 reduces the learning rate over time. Default is 0.5.
+            threshold (float, optional): Amount by which the loss must improve to count as an
+                improvement. Default is 1e-4.
+            min_lr (float, optional): Minimum learning rate past which no further reducing
+                is applied.  Default is 1e-5.
+            verbose (bool, optional): If True, the logger prints when the learning rate
+                decreases or reaches the minimum (INFO level)
+        """
+        super().__init__(on=on, called_every=called_every)
+        self.monitor = monitor
+        self.mode = mode
+        self.patience = patience
+        self.gamma = gamma
+        self.threshold = threshold
+        self.min_lr = min_lr
+        self.verbose = verbose
+
+        self.best_value = float("inf")
+        self.counter = 0
+        self.reached_minimum = False
+
+    def run_callback(self, trainer: Any, config: TrainConfig, writer: BaseWriter) -> None:
+        """
+        Reduces the learning rate when the loss reaches a plateau.
+
+        Args:
+            trainer (Any): The training object.
+            config (TrainConfig): The configuration object.
+            writer (BaseWriter): The writer object for logging.
+        """
+        if not self.reached_minimum:
```

Comment:
Change name of variable for clarity

```suggestion
        if not self.reached_minimum_lr:
```